tests/tensor/test_basic.py

```diff
@@ -4608,3 +4609,11 @@ def core_np(x, y):
         vectorize_pt(x_test, y_test),
         vectorize_np(x_test, y_test),
     )
+
+
+def test_where_for_only_condition():
+    a = np.array([1, 2, 3, 4, 5])
+    cond = a <= 3
+    pt_result = where(cond)[0].eval()
```

Comment:
Don't let eval constrain you. It is just a debug helper.

In this case, like you said, you have a multiple output operation which you can evaluate just fine by definitely a Pytensor function with multiple outputs. eval does the same under the hood but only works for single variables.